kgforge/specializations/models/rdf/directory_service.py

```diff
@@ -15,25 +15,28 @@
 from pathlib import Path
 from typing import Dict, Tuple
 
-from pyshacl import validate
-from rdflib import Graph, URIRef
+from pyshacl import Shape, validate
+import rdflib
+from rdflib import OWL, Graph, Namespace, URIRef
+
 from rdflib.util import guess_format
 
 from kgforge.core.commons.context import Context
+from kgforge.core.commons.sparql_query_builder import build_shacl_query
 from kgforge.specializations.models.rdf.node_properties import NodeProperties
 from kgforge.specializations.models.rdf.service import RdfService, ShapesGraphWrapper
 
 
 class DirectoryService(RdfService):
 
     def __init__(self, dirpath: Path, context_iri: str) -> None:
-        self._graph = load_rdf_files(dirpath)
+
+        self._graph = _load_rdf_files_as_graph(dirpath)
         self._sg = ShapesGraphWrapper(self._graph)
         super().__init__(self._graph, context_iri)
 
     def schema_source_id(self, schema_iri: str) -> str:
-        # FIXME should return the file path where the schema is in
-        return schema_iri
+        return self.shape_to_defining_resource[schema_iri]
```

Comment:
This method is supposed to say where the schema has been defined so it should refer to the path of the file that was loaded that contains the schema

Reply:
I've gone into the details of this in this MR: https://github.com/BlueBrain/nexus-forge/pull/351/files#diff-defd061221b05700c81209720e0efdc6ce1fdb14edc68e93a926d8f1e7af8a6eR59  

But the problem is that in the former "load_rdf_files" (now named _load_rdf_files_as_graph), the record of which file led to which graph should be kept, and used here

Reply:
I still think the current if elif else structure is not very intuitive. It would be better to have the elif and else in a nested else